connectors/es/sink.py

```diff
@@ -599,9 +617,17 @@ async def get_docs(self, generator, skip_unchanged_documents=False):
                         }
                     )
 
+                # We try raising every loop to not miss a moment when
+                # too many errors happened when downloading
+                lazy_downloads.raise_any_exception()
+
                 await asyncio.sleep(0)
+
+            # Sit and wait until an error happens
+            await lazy_downloads.join(raise_on_error=True)
         finally:
             # wait for all downloads to be finished
+            # even if we errored out
             await lazy_downloads.join()
```

Comment:
Since we call `await lazy_downloads.join(raise_on_error=True)` above in the success case, we'd only get down here without having already called `join` if an error was raised. In that case, should we do `lazy_downloads.cancel()`?

Reply:
This logic is here to actually wait for currently running downloads to finish. Alternatively we could cancel,  but even cancellation requires joining in the end

Reply:
So should we cancel and then join?